lib/compress/index.js

```diff
@@ -2174,6 +2178,19 @@ def_optimize(AST_Binary, function(self, compressor) {
                 self.left = make_node(AST_Undefined, self.left).optimize(compressor);
                 if (self.operator.length == 2) self.operator += "=";
             }
+        } else if (compressor.option("typeofs")
```

Comment:
This block of code is only applicable for `comparisons: true, typeofs: true, lhs_constants: false`. I'm not sure if the added complexity is worth it.